### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by Elijahscriptdev**
Milestone 3

### CODE REVIEW

File: lib/player.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  def num_choice(game_board)
+    @choice = gets.to_i
+    until game_board[@choice - 1] == ' ' && @choice.between?(1, 9)
+      puts 'Wrong choice!!!!!!!!!!!'
+      puts 'Please input another number between 1-9 and not used already'
+      @choice = gets.to_i
+    end
+    @choice
+  end
```
Comment (maelfosso) [inline lib/player.rb:29]: no `puts` nor `gets` nor `print` in any logic files

File: lib/player.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class Player
+  attr_accessor :choice, :player, :count
```
Comment (maelfosso) [inline lib/player.rb]: `attr_accessor` is not allowed

File: lib/board.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+class Board
+  attr_accessor :game_board, :board_num
```
Comment (maelfosso) [inline lib/board.rb]: `attr_accessor` is not allowed.
Choose between `attr_reader` or `attr_writer`

File: lib/board.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      return @board_num << i if @game_board[i] == player
+
+      @board_num << nil
+    end
+    board_num
+  end
+
+  def winner?
+    WIN.any? { |line| (line - @board_num) == [] }
+  end
```
Comment (maelfosso) [inline lib/board.rb:33]: Where is WIN ?
Could define it here ?

File: lib/tic_tac_toe.rb [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        puts "Winner player #{@player.player}!"
+        break
+      end
+      @count += 1
+    end
+    return unless game_draw?
+
+    puts 'It is a draw...try again!!!'
+  end
+end
```
Comment (maelfosso) [inline lib/tic_tac_toe.rb:67]: No puts in logic files

File: lib/player.rb (hunk 1 above)
Comment (maelfosso) [reply lib/player.rb:29]: I suggest you to move that method into the Board class. 
There is no reason to keep it in the Player class:
- It doesn't use anything from that class
- The Player class could work without this method

On the other hand, the simple fact that you pass the board, `@board.game_board` from the Board class, as a parameter in that method means that you can have the same method into the Board class  and 
```ruby
def num_choice(choice)
    ...
    until @game_board[choice - 1] == ' ' && choice.between?(1, 9)
      puts 'Wrong choice!!!!!!!!!!!'
      puts 'Please input another number between 1-9 and not used already'
      ...
    end
    ...
  end
```
That is just a mock as an example

File: lib/board.rb (hunk 4 above)
Comment (maelfosso) [reply lib/board.rb:33]: This method is not working!!!

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required

Hello @Elijahscriptdev 

0. Could you please, display the empty board before asking the player to play is first move ?

1. We should be able to run the game with only `bin/main.rb`. I got an error when trying

2. Except for **draw** state, the game doesn't any of winning state, weither is the horizontal, vertical or diagonal one.

3. Read the comments below.

4. Fix the sticklers errors.

Kindly make the changes and submit a new code review request
Thanks

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com) | [LinkedIn](https://www.linkedin.com/in/mael-elvis-fosso-650b6346/)

**Review by uche-inyama — APPROVED** [review]
## Approved

Hi Elijah!

Everything is working fine, go ahead and merge into development.

#### ``` Have a great day ``` :slightly_smiling_face: 

``` Uchechukwu Henry Inyama```

**Pull request merged**